### ISSUE TITLE
move check of camera TTL from state-machine to initialization

### ISSUE THREAD

**Issue by bimac** [int-brain-lab/iblrig#558]
currently, when starting a session inheriting from `ActiveChoiceWorldSession`, the task blocks up if no sync pulses from the camera are received on the Bpod's behavior port during the first state of the state-machine:

https://github.com/int-brain-lab/iblrig/blob/3d16f407397f9ab9342746fe95d0b4a71a5d0212/iblrig/base_choice_world.py#L578-L585

could/should this state have a time-out or be entirely optional?

- [x] adapt state-machine (separate branch)

**Comment by bimac** [comment]
@oliche @samupicard

**Comment by samupicard** [comment]
I don't know about the design question, but the camera-dependency definitely needs to be parametrized. A buggy video PC can currently block up the entire task pipeline, and there is no workaround

**Comment by k1o0** [comment]
> also, from a design-perspective - does it make sense to do this in the state-machine at all?

I don't believe it does. This should be part of the pre-session hardware preparation and not executed during the first trial. Because the length of the first trial start pulse is undefined and the Bpod out is multiplexed, our extractors have to 'guess' which pulse to assign to the first trial start event:

https://github.com/int-brain-lab/ibllib/blob/526385b3e7b3962cbd910911409cbf03c4e5a17d/ibllib/io/extractors/ephys_fpga.py#L1149-L1160

This is very unreliable as the valve pulse is also variable in length (depends on water volume and calibration).  What's more, if the first trial start pulse is missed on the FPGA (e.g. due to the DAQ starting after task starts, https://github.com/int-brain-lab/ibllib/issues/909) then the first valve pulse may be erroneously interpreted as the trial start. All of this could be resolved by leaving the camera trigger out of the state machine altogether. This could be done elsewhere by manually setting Bpod high until the first Port1In event, or maybe with a pre-task state machine.

**Comment by bimac** [comment]
make sure to use specific version tag when testing
